Week3/homework/step2-3.js

```diff
@@ -31,8 +43,15 @@ function repeatStringNumTimesWithDoWhile(str, num) {
   // eslint-disable-next-line
   let result = '';
 
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
+  if (num <= 0) {
```

Comment:
Here, you need to test the condition of `num` before entering the loop as the `do-while` loop condition is tested at the end of the loop. This means that the loop body is executed at least once, without taking the loop condition into account. However, you do not need to re-initialise the `result` variable to the empty string as you already did that in line 44.

The fact that the `do-while` condition is tested at the end of the loop makes this construct rather useless. A `do-while` loop can always be rewritten as regular `while` loop. Using a `do-while` loop is like having unsafe sex on your first date 😁.